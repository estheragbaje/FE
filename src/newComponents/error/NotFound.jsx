import React from 'react';
import { MdKeyboardBackspace } from 'react-icons/md';
import propTypes from 'prop-types';

import { Container } from '../dashboard/DashboardLayoutStyles';
import image from '../assets/images/404.svg';

function NotFound(props) {
  const goHome = (e) => {
    e.preventDefault();
    props.history.push('/');
  };
  return (
    <Container>
      <div>
        <img src={image} alt="Page Not Found" load="lazy" />
      </div>
      <button type="button" onClick={goHome}>
        <MdKeyboardBackspace />
        <span>Home</span>
      </button>
    </Container>
  );
}

NotFound.propTypes = {
  history: propTypes.shape({
    push: propTypes.func.isRequired,
  }).isRequired,
};
export default NotFound;
