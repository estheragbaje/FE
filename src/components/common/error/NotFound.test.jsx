import React from 'react';
import { shallow } from 'enzyme';

import NotFound from './NotFound';

describe('<NotFound />', () => {
  it('renders the NotFound component correctly', () => {
    const wrapper = shallow(<NotFound />);
    expect(wrapper).toMatchSnapshot();
  });
});
