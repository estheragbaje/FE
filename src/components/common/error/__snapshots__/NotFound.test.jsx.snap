// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders the NotFound component correctly 1`] = `
<styled.div>
  <div>
    <img
      alt="Page Not Found"
      load="lazy"
      src="404.svg"
    />
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    <MdKeyboardBackspace />
    <span>
      Home
    </span>
  </button>
</styled.div>
`;
